### PR TITLE
Fixes #10828 - facts are logged when discovery fails

### PR DESCRIPTION
**Pull request by lzap**
This is WIP patch to improve our logging and make use of our new logging
capabilities. I will do this work in two separate patches, the first one will
improve logging in a compatible way with older versions (can be cherry picked)
and the latter will introduce support for plugin logger via logger gem.

### CODE REVIEW

File: app/controllers/api/v2/discovered_hosts_controller.rb [hunk 1]
```diff
@@ -110,6 +110,7 @@ def facts
         end
         process_response state
       rescue ::Foreman::Exception => e
```
Comment (lzap) [inline app/controllers/api/v2/discovered_hosts_controller.rb]: This should be `Exception => e` instead!

File: app/controllers/api/v2/discovered_hosts_controller.rb [hunk 2]
```diff
@@ -110,6 +110,7 @@ def facts
...
+        logger.warn "Host discovery failed, facts: #{params[:facts]}"
         render :json => {'message'=>e.to_s}, :status => :unprocessable_entity
```
Comment (lzap) [inline app/controllers/api/v2/discovered_hosts_controller.rb:115]: We should print stacktrace in logs here.

**Comment by stbenjam** [comment]
[test]

**Comment by stbenjam** [comment]
Is this still WIP?

**Comment by lzap** [comment]
Oh yeah, let me finish that.

**Comment by lzap** [comment]
Pushed, note discovery still does not use the new logging capabilities. I want this to be merged first so I can backport it and the new logging will come later this week. Thanks!

**Comment by stbenjam** [comment]
Merged as 15c1c9d9a5652478cc8df059443d831ed8abfc4a, thanks @lzap!

**Pull request closed without merging**